src/matrixeventprocessor.ts

```diff
@@ -123,21 +142,23 @@ export class MatrixEventProcessor {
                     await this.ProcessMsgEvent(event, srvChanPair[0], srvChanPair[1]);
                     return;
                 } catch (err) {
-                    log.warn("There was an error sending a matrix event", err);
-                    return;
+                    throw wrap(err, Error, "There was an error sending a matrix event");
```

Comment:
`wrap` is not nearly clear enough as a name. What are you wrapping?

Reply:
I kind of like the name. It is wrapping the original error into a new one (while preserving the stacktrace).

Reply:
A low-level `IOError` at a higher level is more likely a `RequestError` or `UserRegistrationError`. That's why it is nice to use this functionality: It gives more meaning to an error.